diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0998bb78..6c7232fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,10 +106,10 @@ if("${FLUX_CORE_LIBDIR}" MATCHES "^${FLUX_CORE_PREFIX}/(.+)$")
 else()
   set(FLUX_CORE_LIBEXT "${CMAKE_INSTALL_LIBDIR}")
 endif()
-set(CMAKE_INSTALL_LIBDIR ${FLUX_CORE_LIBEXT} CACHE PATH "..." FORCE)
+set(CMAKE_INSTALL_LIBDIR ${FLUX_CORE_LIBEXT} CACHE STRING "..." FORCE)
 message(STATUS "Forcing libdir selection to match flux-core: prefix subdir (${FLUX_CORE_LIBEXT})")
 message(STATUS "    flux-core library path: (${FLUX_CORE_LIBDIR})")
-message(STATUS "    flux-sched library path: (${CMAKE_INSTALL_PREFIX}/${FLUX_CORE_LIBEXT})")
+message(STATUS "    flux-sched library path: (${CMAKE_INSTALL_FULL_LIBDIR})")
 if(DEFINED ENV{PYTHON})
   set(Python_EXECUTABLE $ENV{PYTHON} CACHE FILEPATH "")
 endif()
@@ -145,7 +145,7 @@ message(STATUS "Boost version: ${Boost_VERSION}")
 
 # Install paths
 set(FLUX_CMD_DIR "${CMAKE_INSTALL_LIBEXECDIR}/flux/cmd")
-set(FLUX_LIB_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/flux")
+set(FLUX_LIB_DIR "${CMAKE_INSTALL_LIBDIR}/flux")
 set(FLUX_MOD_DIR "${FLUX_LIB_DIR}/modules")
 set(FLUX_SHELL_PLUGIN_DIR "${FLUX_LIB_DIR}/shell/plugins")
 set(FLUX_RC1_DIR "${CMAKE_INSTALL_SYSCONFDIR}/flux/rc1.d")
