diff --git a/t/t0001-basic.t b/t/t0001-basic.t
index f2982d1f5..d4c9d9e64 100755
--- a/t/t0001-basic.t
+++ b/t/t0001-basic.t
@@ -697,7 +697,7 @@ test_expect_success HAVE_MAN 'flux-help command can display manpages for subcomm
 	.SH NAME
 	foo \- foo bar baz
 	EOF
-	MANPATH=${PWD}/man FLUX_IGNORE_NO_DOCS=y flux help foo | grep "^FOO(1)"
+	MANPATH=${PWD}/man FLUX_IGNORE_NO_DOCS=y flux help foo | grep "Foo utils"
 '
 test_expect_success HAVE_MAN 'flux-help command can display manpages for api calls' '
 	PWD=$(pwd) &&
@@ -707,7 +707,7 @@ test_expect_success HAVE_MAN 'flux-help command can display manpages for api cal
 	.SH NAME
 	flux_foo \- Call the flux_foo interface
 	EOF
-	MANPATH=${PWD}/man FLUX_IGNORE_NO_DOCS=y flux help flux_foo | grep "^FOO(3)"
+	MANPATH=${PWD}/man FLUX_IGNORE_NO_DOCS=y flux help flux_foo | grep "Foo api call"
 '
 missing_man_code()
 {
